### PR TITLE
[DocBlock] Update docblock contents right in the rule

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to https://github.com/rectorphp/rector-src/pull/4987

### CODE REVIEW

File: tests/Issues/PartialValueDocblockUpdate/Source/PartialUpdateTestRector.php [hunk 1]
```diff
@@ -42,6 +48,8 @@ public function refactor(Node $node)
                 'defaults',
             );
 
+            $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
```
Comment (samsonasik) [inline tests/Issues/PartialValueDocblockUpdate/Source/PartialUpdateTestRector.php:51]: /cc @shyim FYI, ref issue and PR:

- https://github.com/rectorphp/rector/issues/7020
- https://github.com/rectorphp/rector-src/pull/1862

File: rules/Php80/DocBlock/PropertyPromotionDocBlockMerger.php [hunk 2]
```diff
@@ -51,6 +53,8 @@ public function mergePropertyAndParamDocBlocks(
 
                 $param->setAttribute(AttributeKey::COMMENTS, $mergedComments);
             }
+        } else {
+            $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($param);
         }
```
Comment (samsonasik) [inline rules/Php80/DocBlock/PropertyPromotionDocBlockMerger.php]: above on after `$param->setAttribute(AttributeKey::COMMENTS, $mergedComments);` may probably need to use docblockupdater as well on line 54

or remove the else, direclty use:

```suggestion
        }
        
        $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($param);
```

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php [hunk 3]
```diff
@@ -121,9 +124,12 @@ public function refactor(Node $node): ?Node
                 $param,
                 $paramName
             );
+
+            $hasChanged = true;
         }
 
-        if ($functionLikePhpDocInfo->hasChanged()) {
+        if ($hasChanged) {
+            $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php]: this should not needed, as already called internally while on `$this->phpDocTypeChanger->changeParamType()` above:

https://github.com/rectorphp/rector-src/blob/596bbc98a25903c525914b4d16ea5f9c76b500fb/packages/BetterPhpDocParser/PhpDocManipulator/PhpDocTypeChanger.php#L176

Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php]: @samsonasik I won't have time tonight to work on this, but I don't want it to be a blocker.
 Could you take this PR over and push the merge when its green? :)

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php]: I will try

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by TomasVotruba** [comment]
Thank you :partying_face: :+1:

**Pull request merged**